services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/AsyncBufferingSubscriber.java

```diff
@@ -40,7 +40,7 @@ public class AsyncBufferingSubscriber<T> implements Subscriber<T> {
     private final int maxConcurrentExecutions;
     private final AtomicInteger numRequestsInFlight;
     private volatile boolean upstreamDone;
-    private Subscription subscription;
+    private volatile Subscription subscription;
```

Comment:
Changed to volatile to make Spotbugs happy. It was complaining about unsynchronized access subscription `onSubscribe`

Reply:
Maybe we can safely ignore the violation (mark it as false positive) since we know onSubscribe is always invoked first?

Reply:
Discussed offline, agreed it's okay since the alternative is suppressing the spotbugs rule for this class which is more risky.